addons/payment/controllers/portal.py

```diff
@@ -175,6 +175,19 @@ def payment_method(self, **kwargs):
         }
         return request.render('payment.payment_methods', rendering_context)
 
+    @http.route('/my/payment_method/archive', type='json', auth='user')
+    def archive_token(self, token_id):
+        user_partner = request.env.user.partner_id
+        token_sudo = request.env['payment.token'].sudo().search([
+            ('id', '=', token_id),
+            ('partner_id', 'in', [user_partner.id, user_partner.sudo().commercial_partner_id.id])
```

Comment:
Now, we clearly see that the `sudo()` is useless and misleading. Also, one could think that a `browse` could do the job so let's add an explanatory comment.
```suggestion
            # Check that the user owns the token before letting them archive anything
            ('partner_id', 'in', [partner_sudo.id, partner_sudo.commercial_partner_id.id])
```